### PR TITLE
Scan oversized lines by chunks

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Add chunk-based scan for oversized lines with overlap to do not skip a credential between chunks
- slowdown ~26% (in the full dataset)
- slowdown ~ 2.2% for dataset without ovesized lines

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -22,7 +22,12 @@ jobs:
       - name: Checkout CredData
         uses: actions/checkout@v4
         with:
-          repository: Samsung/CredData
+          repository: babenek/CredData
+          ref: tachunks
+
+      - name: Markup hashing
+        run: |
+          for f in $(find meta -type f|sort); do md5sum $f; done >meta.md5
```
Comment (babenek) [inline .github/workflows/benchmark.yml:30]:
```suggestion
```

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -66,7 +71,12 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:79]:
```suggestion
```

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -171,7 +181,12 @@ jobs:
...
```
Comment (babenek) [inline .github/workflows/benchmark.yml:189]:
```suggestion
```

File: .github/workflows/benchmark.yml [hunk 4]
```diff
@@ -348,7 +363,12 @@ jobs:
       - name: Checkout CredData
         uses: actions/checkout@v3
         with:
-          repository: Samsung/CredData
+          repository: babenek/CredData
+          ref: tachunks
+
+      - name: Markup hashing
+        run: |
+          for f in $(find meta -type f|sort); do md5sum $f; done >meta.md5
```
Comment (babenek) [inline .github/workflows/benchmark.yml:371]:
```suggestion
```

**Review by babenek — COMMENTED** [review]
rollback

**Review by csh519 — COMMENTED** [review]
Thank you for suggest great idea to scanning oversize lines!
It seems there is a problem when scanning full benchmark data...
So how about set oversize limit const like as below?

``` python
OVERSIZE_LINE_LIMIT = MAX_LINE_LENGTH * 10  # You can change the number.
```

I think it would be good to find a number which is 10 above, by testing with benchmark data. I hope the slowdown should under 10%..

**Comment by babenek** [comment]
todo: rollback custom BM ref after https://github.com/Samsung/CredData/pull/142

**Comment by babenek** [comment]
@csh519 
The overlap duplicates issue is fixed.
https://github.com/Samsung/CredSweeper/pull/522/files#diff-a308e37dd0360462596f1887de9d165bf5fb16132a748481cc54a1b64b77b62f
- two the same Candidates were used in the test. The purging is processed with unique positions of value, variable, separator and line.

**Review by csh519 — DISMISSED** [review]
Oversize lines can be scanned after this PR.
Thank you for suggestion!

LGTM :+1:

**Review by xDizzix — APPROVED** [review]

**Review by iuriimet — APPROVED** [review]

**Pull request merged**